### PR TITLE
cleaning up and fixing links

### PR DESCRIPTION
**Pull request by sthetland**
This PR makes the format of the links consistent to the prevailing style in the docs and fixes a few broken external links. 

Using .md in link text rather than .html causes Docusarus to convert the links into absolute links rather than using relative links, making the docs easier to port to other contexts. 

Also updating a few broken external links.

### CODE REVIEW

**Review by suneet-s — APPROVED** [review]
LGTM

Is there a script you ran to find all these broken links? We could add this to the travis job that runs on every PR to make sure we don't break them in the future

**Comment by sthetland** [comment]
Thanks @suneet-s.  I didn't use a script, no. I found these by search & replace, since each one required manual confirmation. I think it may be possible to automate, if we could come up with a clever enough regex, but I didn't get there while making these changes. Maybe be worth looking into.

**Comment by suneet-s** [comment]
Thanks for the contribution @sthetland !

File: docs/querying/caching.md [hunk 1]
```diff
@@ -82,17 +82,3 @@ Note that the task executor processes only support caches that keep their data l
 This restriction exists because the cache stores results at the level of intermediate partial segments generated by the
 ingestion tasks. These intermediate partial segments will not necessarily be identical across task replicas, so
 remote cache types such as `memcached` will be ignored by task executor processes.
-
-## Unsupported queries
```
Comment (abhishekagarwal87) [inline docs/querying/caching.md:86]: Hi @sthetland  what was the reason for removing this para in this PR? was it accidental?

**Pull request merged**